.github/workflows/train.yml

```
@@ -19,43 +20,72 @@ jobs:
+        env:
+          PROJECT_ID: engineeringlab
+          APPLICATION_CREDENTIALS: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
+        with:
+          args: auth configure-docker gcr.io --quiet
+      - name: Login to GCR
+        uses: docker/login-action@v1
         with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          registry: gcr.io
+          username: _json_key
+          password: ${{ secrets.GCR_JSON_KEY }}
       - name: Build and push
-        id: docker_build
         uses: docker/build-push-action@v2
         with:
-          push: true
           context: ./Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier
-          file: ./Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile
-          tags: englabsteam1/trainer:latest
+          file: ./Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/Dockerfile-trainer
+          push: true
+          tags: gcr.io/engineeringlab/trainer:${{ github.sha }}
           build-args: |
             GCP_CREDS_JSON_BASE64=${{ secrets.GCP_CREDS_JSON_BASE64 }}
-      # Alternative manual build (for comparison, only one branch will be left)
-      - name: Build trainer image
-        run: |
-          cd Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier && \
-          docker build -t engineering-labs1/trainer \
-          --build-arg GCP_CREDS_JSON_BASE64="${GCP_CREDS_JSON_BASE64}" \
-          -f Dockerfile-trainer .
-      - name: Deploy trainer and run training
-        run: |
-          echo TODO
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+      - name: Start training node
+        uses: actions-hub/gcloud@master
+        env:
+          PROJECT_ID: engineeringlab
+          APPLICATION_CREDENTIALS: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
+        with:
+          args: |
+            compute instances start --zone us-east1-b compute-1
+      - name: Sleep for 15 seconds
+        uses: jakejarvis/wait-action@master
+        with:
+          time: '15s'
+      - name: Run proper training
+        uses: actions-hub/gcloud@master
+        env:
+          PROJECT_ID: engineeringlab
+          APPLICATION_CREDENTIALS: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
+          MLFLOW_TRACKING_URI: http://35.185.111.8:5000
+        with:
+          args: |
+            compute ssh --zone us-east1-b compute-1 -- 'export TMPDIR=$(mktemp -d /tmp/model.XXXXXXXXXX) && cd $TMPDIR && git clone https://github.com/mlops-labs-team1/engineering.labs.git && cd engineering.labs/Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/ && git checkout an/gh-actions && sed -i -e s/##IMAGE##/${{ github.sha }}/ MLproject && export JSON_FILE=${{ github.sha }}.json && export MLFLOW_TRACKING_URI='${MLFLOW_TRACKING_URI}' &&  mlflow run --no-conda -P json_dump=/tmp/englab/$JSON_FILE .'
```

If you change the MLProject default
```suggestion
            compute ssh --zone us-east1-b compute-1 -- 'export TMPDIR=$(mktemp -d /tmp/model.XXXXXXXXXX) && cd $TMPDIR && git clone https://github.com/mlops-labs-team1/engineering.labs.git && cd engineering.labs/Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert-classifier/ && git checkout an/gh-actions && sed -i -e s/##IMAGE##/${{ github.sha }}/ MLproject && export JSON_FILE=${{ github.sha }}.json && export MLFLOW_TRACKING_URI='${MLFLOW_TRACKING_URI}' &&  mlflow run --no-conda -P json_dump=/tmp/englab/$JSON_FILE -P model_name=BertModel .'
```